packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1406,36 +1618,42 @@ typedef T ComputePropertyValueCallback<T>();
 /// If the default `value.toString()` does not provide an adequate description
 /// of the value, specify `description` defining a custom description.
 ///
-/// The [hidden] property specifies whether the property should be hidden from
-/// the default output (e.g. the output given by [toStringDeep]).
-///
 /// The [showSeparator] property indicates whether a separator should be placed
 /// between the property [name] and its [value].
 class DiagnosticsProperty<T> extends DiagnosticsNode {
   /// Create a diagnostics property.
   ///
-  /// The [hidden], [showName], [showSeparator], and [style] arguments must not be null.
+  /// The [showName], [showSeparator], [style], and [missingIfNull] arguments
+  /// must not be null.
+  ///
+  /// At most one of `hidden` and [level] should be specified. Setting `hidden`
+  /// to `true` is equivalent to setting [level] to [DiagnosticLevel.hidden].
+  /// If neither `hidden` or [level] is set, [level] defaults to
+  /// [DiagnosticsLevel.info].
   DiagnosticsProperty(
     String name,
     T value, {
     String description,
-    bool hidden: false,
```

Comment:
you've already broken back-compat by removing the "hidden" property getter, I'd just go all the way and remove the argument too. I don't think it provides much value.

Reply:
Removed the hidden argument. Most of the cases where it was used were IterableProperty cases better handled by `ifEmpty: null` which I've added support for.